.github/workflows/hosted-pure-workflow.yml

```diff
@@ -86,15 +86,26 @@ jobs:
       # On Windows runners, let's ensure to have the Developer Command Prompt environment setup correctly.
       # As used here the Developer Command Prompt created is targeting x64 and using the default the Windows SDK.
       - uses: ilammy/msvc-dev-cmd@v1
-
       # Run CMake to generate Ninja project files, using the vcpkg's toolchain file to resolve and install
       # the dependencies as specified in vcpkg.json. Note that the vcpkg's toolchain is specified
       # in the CMakePresets.json file.
       # This step also runs vcpkg with Binary Caching leveraging GitHub Action cache to
       # store the built packages artifacts.
       - name: Restore from cache the dependencies and generate project files
         run: |
-          cmake --preset ninja-multi-vcpkg
+          cmake --preset ninja-multi-vcpkg -DCMAKE_EXPORT_COMPILE_COMMANDS=ON
```

Comment:
are the compile commands useful? 
and if they are, docs should be added.
But my idea is to keep this template minimal, hence if there are no strong reasons, i'd drop it

Reply:
`DCMAKE_EXPORT_COMPILE_COMMANDS` is necessary for clang-tidy to work. 
I've moved this flag to CMakeLists.txt to remove the duplication.

Reply:
Since it is necessary for clang-tidy, looks good to me, thanks.